content/blueprints/gcp.md

```diff
@@ -0,0 +1,1277 @@
+
+
+
+### Example
+
+{{< gsHighlight  yaml  >}}
+
+my_gcp_instance:
+  type: cloudify.gcp.nodes.Instance
+  properties:
+    image_id: http://url.to.your.example.com/image
+    instance_type: n1-standard-1
+    gcp_config:
+      project: your-project
+      network: default
+      zone: us-east1-b
+      auth: path_to_auth_file.json
+
+{{< /gsHighlight >}}
+
+This example includes shows adding additional parameters, tagging an instance name, and explicitly defining the gcp_config.
+
+
+
+
+## cloudify.gcp.nodes.InstanceGroup
+**Derived From:** [cloudify.nodes.Root]({{< relref "blueprints/built-in-types.md" >}})
+
+A GCP InstanceGroup.
+This is used to configure failover systems. InstanceGroups can be configured to scale automatically based on load, and will replace failing Instances with freashly started ones.
+
+
+**Properties:**
+
+
+  * `gcp_config`
+    A dictionary of values to pass to authenticate with the GCP API.
+
+    *default:* {}
+  * `additional_settings`
+    Additional setting for instance group
+
+    *default:* {}
+  * `use_external_resource`
+    Indicate whether the resource exists and use existing (true) or if Cloudify should create new resource (false).
+
+    *type:* boolean
+    *default:* False
+  * `name`
+    Optional instance name. By default it will be instance group id.
+
+    *type:* string
+    *default:*
+  * `named_ports`
+    A list of named ports defined for this instance group, the expected format is: [{name: 'name', port: 1234}, ... ].
+
+    *default:* []
+
+
+
+
+
+## cloudify.gcp.nodes.KeyPair
+**Derived From:** [cloudify.nodes.Root]({{< relref "blueprints/built-in-types.md" >}})
+
+An SSH key-pair which will be uploaded to any Instances connected to it via `cloudify.gcp.relationships.instance_connected_to_keypair`.
+
+Unlike other cloud providers, users are dynamically created on Instances based on the username specified by the uploaded SSH key, so the public key text must include a username in the comment section (keys generated using `ssh-keygen` have this by default).
+
+
+
+**Properties:**
+
+
+  * `private_key_path`
+    The path where the key should be saved on the machine. If this is a bootstrap process, this refers to the local computer. If this will run on the manager, this will be saved on the manager.
+
+    *type:* string
+    *default:*
+  * `public_key_path`
+    The path to read from existing public key.
+
+    *type:* string
+    *default:*
+  * `user`
+    The user account for this key. A corresponding user account will be created by GCP when the key is added to the Instance. This must be supplied for a non-external resource key. See https://cloud.google.com/compute/docs/instances/adding-removing-ssh-keys
+
+    *type:* string
+    *default:*
+  * `gcp_config`
+    A dictionary of values to pass to authenticate with the Google Cloud Platform API.
+
+    *default:* {}
+  * `use_external_resource`
+    Indicate whether the resource exists or if Cloudify should create the resource.
```

Comment:
Indicate**s** whether the resource exists **and should be used (true),** or if Cloudify should create the resource **(false).**